### DISCUSSION TITLE
Klein probleem met installeren.

### DISCUSSION THREAD

**Discussion by sec-ed** [minvws/nl-kat-coordination#14] [General]
Hi,
Eindelijk heb ik de tijd gevonden om openKAT te installeren.

Ik loop echter vast op het klonen van de repository.
"
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
"
Blijkbaar heb ik niet de juiste rechten?!
Hoe fix ik dit?

**Comment by reinschaap** [comment]
https://github.com/minvws/nl-kat-coordination/wiki/Installatiehandleiding-KAT#41-standaard-installatie

Deze stap lukt je niet?

De repo is publiekelijk dus daar zitten geen toegangs limieten op

**Comment by iwanVgeelen** [comment]
Rein is sneller :-)

Probeer jij naar de nieuwe Git? 
$ git clone git@github.com:minvws/nl-kat-coordination.git
Of de vorige?

**Comment by confiks** [comment]
Github laat het niet toe om te pullen via SSH zonder gebruikersaccount en bijbehorende SSH key.

Probeer in plaats daarvan `git clone https://github.com/minvws/nl-kat-coordination.git`

**Comment by sec-ed** [comment]
Bedankt voor de reacties.
Mijn verbazing was even groot over de melding.
Omdat ik inderdaad naar de nieuwe/openbare ging.

Ik had de locatie vanuit de handleiding gekopieerd, en dat is de juiste.
"
~$ git clone git@github.com:minvws/nl-kat-coordination.git
Cloning into 'nl-kat-coordination'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
"

Gisteren, met enige frustratie, mijn notebook dichtgeklapt.
Vandaag, als eerste opdracht, zonder dat ik iets heb aangepast, nogmaals geprobeerd en.. 
"
~$ git clone https://github.com/minvws/nl-kat-coordination.git
Cloning into 'nl-kat-coordination'...
remote: Enumerating objects: 46, done.
remote: Counting objects: 100% (46/46), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 46 (delta 14), reused 24 (delta 4), pack-reused 0
Receiving objects: 100% (46/46), 24.42 KiB | 1.53 MiB/s, done.
Resolving deltas: 100% (14/14), done.
"

Beetje raadselachtig, ik ga er verder ook maar niet te veel over nadenken.

**Comment by sec-ed** [comment]
Helaas, volgende stap geeft nieuwe problemen.
(Ik heb trouwens een nieuwe virtual-box met ubuntuLTS voor openKAT opgetuigd.)

~/nl-kat-coordination$ make kat
make clean
make clone
make build
make up
make[1]: Entering directory '/home/user/nl-kat-coordination'
docker-compose down
docker volume rm nl-kat_rocky-db-data nl-kat_bytes-db-data nl-kat_katalogus-db-data
ERROR: build path /home/user/nl-kat-coordination/nl-kat-octopoes either does not exist, is not accessible, or is not a valid URL.
Error: No such volume: nl-kat_rocky-db-data
Error: No such volume: nl-kat_bytes-db-data
Error: No such volume: nl-kat_katalogus-db-data
make[1]: [Makefile:42: clean] Error 1 (ignored)
make[1]: Leaving directory '/home/user/nl-kat-coordination'
make[1]: Entering directory '/home/user/nl-kat-coordination'
git clone git@github.com:minvws/nl-kat-boefjes.git
git clone git@github.com:minvws/nl-kat-bytes.git
git clone git@github.com:minvws/nl-kat-octopoes.git
git clone git@github.com:minvws/nl-kat-mula.git
git clone git@github.com:minvws/nl-kat-rocky.git
Cloning into 'nl-kat-boefjes'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'nl-kat-bytes'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'nl-kat-octopoes'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'nl-kat-mula'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'nl-kat-rocky'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
make[1]: [Makefile:49: clone] Error 128 (ignored)
make[1]: Leaving directory '/home/user/nl-kat-coordination'
make[1]: Entering directory '/home/user/nl-kat-coordination'
docker-compose build
docker-compose run --rm rocky make build
make -C nl-kat-boefjes build
make -C nl-kat-bytes build
ERROR: build path /home/user/nl-kat-coordination/nl-kat-octopoes either does not exist, is not accessible, or is not a valid URL.
ERROR: build path /home/user/nl-kat-coordination/nl-kat-octopoes either does not exist, is not accessible, or is not a valid URL.
make[2]: Entering directory '/home/user/nl-kat-coordination'
make[2]: *** nl-kat-boefjes: No such file or directory.  Stop.
make[2]: Leaving directory '/home/user/nl-kat-coordination'
make[2]: Entering directory '/home/user/nl-kat-coordination'
make[2]: *** nl-kat-bytes: No such file or directory.  Stop.
make[2]: Leaving directory '/home/user/nl-kat-coordination'
make[1]: *** [Makefile:93: build] Error 2
make[1]: Leaving directory '/home/user/nl-kat-coordination'
make[1]: Entering directory '/home/user/nl-kat-coordination'
docker-compose up -d --force-recreate rocky
ERROR: build path /home/user/nl-kat-coordination/nl-kat-octopoes either does not exist, is not accessible, or is not a valid URL.
make[1]: *** [Makefile:46: up] Error 1
make[1]: Leaving directory '/home/user/nl-kat-coordination'
make: *** [Makefile:21: kat] Error 2

**Comment by confiks** [comment]
Ook in de Makefile wordt `git clone` met SSH URLs gebruikt in plaats van HTTPS. Aangezien je geen SSH key bij Github hebt geregistreerd, gaat dit ook fout (zie documentatie [hier](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)). Ik heb een [PR gemaakt](https://github.com/minvws/nl-kat-coordination/pull/15), maar je kan het ook lokaal fixen.

**Reply by sec-ed** [reply]
Thanks!!
Heb een ssh key geregistreerd en toen ging het verder zonder problemen.
Katje zit nu met zijn eerste bolletje wol te spelen in afwachting van de eerste scan. :)
